api/infrastructure/src/unit/delete.rs

```diff
@@ -0,0 +1,32 @@
+use crate::ServerState;
+use diesel::prelude::*;
+use rocket::http::Status;
+
+pub fn db_delete_unit(
+    unit_code: &str,
+    state: &ServerState,
+) -> Result<Option<String>, (Status, Option<String>)> {
```

Comment:
Is there a reason we return an `Option<String>` here when we only ever return `Ok(None)`?